bpdm-gate-api/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/api/client/GateClientImpl.kt

```diff
@@ -38,26 +38,18 @@ class GateClientImpl(
             .build()
     }
 
-    private val gateClientBusinessPartner by lazy { httpServiceProxyFactory.createClient(GateBusinessPartnerApi::class.java) }
-    private val gateClientAddress by lazy { httpServiceProxyFactory.createClient(GateAddressApi::class.java) }
-    private val gateClientLegalEntity by lazy { httpServiceProxyFactory.createClient(GateLegalEntityApi::class.java) }
-    private val gateClientSite by lazy { httpServiceProxyFactory.createClient(GateSiteApi::class.java) }
-    private val gateClientChangelog by lazy { httpServiceProxyFactory.createClient(GateChangelogApi::class.java) }
-    private val gateClientSharingState by lazy { httpServiceProxyFactory.createClient(GateSharingStateApi::class.java) }
+    override val businessParters by lazy { createClient<GateBusinessPartnerApi>() }
 
-    override fun businessParters() = gateClientBusinessPartner
+    override val addresses by lazy { createClient<GateAddressApi>() }
 
-    override fun addresses() = gateClientAddress
+    override val legalEntities by lazy { createClient<GateLegalEntityApi>() }
 
-    override fun legalEntities() = gateClientLegalEntity
+    override val sites by lazy { createClient<GateSiteApi>() }
 
-    override fun sites() = gateClientSite
+    override val changelog by lazy { createClient<GateChangelogApi>() }
 
-    override fun changelog() = gateClientChangelog
-
-    override fun sharingState() = gateClientSharingState
+    override val sharingState by lazy { createClient<GateSharingStateApi>() }
 
+    private inline fun <reified T> createClient() =
+        httpServiceProxyFactory.createClient(T::class.java)
```

Comment:
I think, tis line needs a bit of documentation. It not so clear, what the httpServiceProxyFactory does.
Is it neccessary to create the client on every call, or could we cache the result somewhere?

Reply:
A client is not created on every call because these properties, e.g. `businessParters`, are lazy properties which are initialized on first access using the lambda function ` { createClient<GateBusinessPartnerApi>() }`.
The only thing I changed here is removing the unnecessary functions and moving the client-creation into an internal helper function.
I agree, some documentation might clarify things, both the old and the new version.

> I think, tis line needs a bit of documentation. It not so clear, what the httpServiceProxyFactory does. Is it neccessary to create the client on every call, or could we cache the result somewhere?

Reply:
Together with Rainer I improved the documentation.